lib/tizen_device.dart

```diff
@@ -210,16 +211,53 @@ class TizenDevice extends Device {
   /// Source: [AndroidDevice.installApp] in `android_device.dart`
   @override
   Future<bool> installApp(TizenTpk app, {String userIdentifier}) async {
+    final bool wasInstalled = await isAppInstalled(app);
+    if (wasInstalled) {
+      if (await isLatestBuildInstalled(app)) {
+        _logger.printStatus('Latest build already installed.');
+        return true;
+      }
+    }
+    final bool isTvEmulator = usesSecureProtocol && await isLocalEmulator;
+    _logger.printTrace('Installing TPK.');
+    if (await _installApp(app, installTwice: !wasInstalled && isTvEmulator)) {
+      return true;
+    }
+    _logger.printTrace('Warning: Failed to install TPK.');
+    if (!wasInstalled) {
+      return false;
+    }
+    _logger.printStatus('Uninstalling old version...');
+    if (!await uninstallApp(app)) {
+      _logger.printError('Error: Uninstalling old version failed.');
+      return false;
+    }
+    if (!await _installApp(app, installTwice: isTvEmulator)) {
```

Comment:
Add a trace log for this second try

Reply:
When the first try fails the user will see

```
Installing TPK.
Installing xxxxxxx.tpk...
Installing TPK failed: (error log)
Warning: Failed to install TPK.
Uninstalling old version...
Installing xxxxxxx.tpk...
```

Isn't this information enough?